app/src/main/java/org/connectbot/service/TerminalManager.java

```diff
@@ -711,7 +720,43 @@ private void reconnectPending() {
 				if (bridge == null) {
 					continue;
 				}
-				bridge.startConnection();
+
+				int sleepVal = 500;
+				int retryLimit = getRetries();
+				int retryCount = 0;
+
+				try {
+					Thread.sleep(sleepVal);
+				} catch(InterruptedException ex) {
+					Thread.currentThread().interrupt();
+				}
+
+				while ( ( retryLimit == 0 || retryCount <= retryLimit ) && bridge.isDisconnected() ){
+
+					bridge.startConnection();
+
+					if ( retryCount < 8 ) {
+						sleepVal = sleepVal * 2;
+					}
+
+					try {
+						Thread.sleep(sleepVal);
+					} catch(InterruptedException ex) {
+						Thread.currentThread().interrupt();
+					}
+
+					retryCount++;
+					Log.d(TAG, String.format("Retries %s session stat: %s", retryCount, bridge.isSessionOpen() ));
+
+					while(bridge.isConnecting()){
+						try {
+							Thread.sleep(500);
```

Comment:
Using minimum retry interval here because it remains constant and this is to wait on the SSH connect which should be more of a known value and not exceed ~30 sec.  The sleepVal grows to address more general network issues which may not as quickly resolve.